k4e-agent-sti-upgrade.sh

```diff
@@ -0,0 +1,194 @@
+This script will create an image file and ISO to be served for download.
+OPTIONS:
+   -h      Show this message
+   -a      Image server address
+   -i      Original Image name(required)
+   -o      Operator's HTTP API (required)
+   -v      Verbose
+EOF
+}
+
+while getopts "h:a:i:o:v" option; do
+    case "${option}"
+    in
+        h)
+            usage
+            exit 0
+            ;;
+        a) IMAGE_SERVER_ADDRESS=${OPTARG};;
+        i) IMAGE_NAME=${OPTARG};;
+        o) HTTP_API=${OPTARG};;
+        v) VERBOSE=1;;
+    esac
+done
+
+dnf install -y bind-utils
+if [[ -z $IMAGE_SERVER_ADDRESS ]]; then
+    hostname_output=$(host $(hostname))
+    for word in $hostname_output
+    do
+      if [[ ($word =~ ^[0-9]+\.[0-9]+\.[0-9]+\.[0-9]+$)  && ($word != "127.0.0.1")]]; then
+        echo "found hostname"
+        IMAGE_SERVER_ADDRESS=$word
+        break
+      fi
+    done
+    if [[ -z $IMAGE_SERVER_ADDRESS ]]; then
+      echo "ERROR: server address is required"
+      usage
+      exit 1
+    fi
+fi
+if [[ -z $IMAGE_NAME ]]; then
+    echo "ERROR: Image name is required"
+    usage
+    exit 1
+fi
+
+if [[ -z $HTTP_API ]]; then
+    echo "ERROR: Operator's HTTP-API url is required"
+    usage
+    exit 1
+fi
+
+if [[ ! -z $VERBOSE ]]; then
+    echo "Building ${IMAGE_NAME} image for Operator's HTTP-API ${HTTP_API}"
+    set -xv
+fi
+
+# export for use in templates
+export IMAGE_NAME
+export HTTP_API
+export RAND_SUFFIX=$((1 + $RANDOM % 10))
+PACKAGES_REPO=k4e-repo-$RAND_SUFFIX
+PACKAGE_REPO_URL=http://$IMAGE_SERVER_ADDRESS/$PACKAGES_REPO
+BLUEPRINT_NAME=$IMAGE_NAME
+IMAGE_UPGRADE_FOLDER=/var/www/html/$IMAGE_NAME-upgrade
+IMAGE_UPGRADE_URL=http://$IMAGE_SERVER_ADDRESS/$IMAGE_NAME-upgrade
+
+#---------------------------------
+# Cleanup before running
+#---------------------------------
+rm -rf ~/rpmbuild/*
+rm -rf /var/www/html/k4e-repo*
+rm -rf /home/builder/yggdrasil
+rm -rf /home/builder/k4e-device-worker
+rm -rf /home/builder/prometheus-node_exporter-rpm
+rm -rf /var/www/html/$IMAGE_UPGRADE_FOLDER
+rm -rf /var/www/html/$IMAGE_NAME-upgrade
+# need to remove all sources
+composer-cli sources delete agent
+
+#-----------------------------------
+# Build packages for k4e from source (change here the branches)
+#-----------------------------------
+# Build yggdrasil rpm
+git clone https://github.com/jakub-dzon/yggdrasil.git /home/builder/yggdrasil
+cd /home/builder/yggdrasil
+export CGO_ENABLED=0
+
+export ARCH=$(uname -i)
+GOPROXY=proxy.golang.org,direct PWD=$PWD spec=$PWD outdir=$PWD make -f .copr/Makefile srpm
+rpm -ihv $(ls -ltr yggdrasil-*.src.rpm | tail -n 1 | awk '{print $NF}')
+if [ "$ARCH" = "aarch64" ]; then
+    # Turn ELF binary stripping off in %post
+    sed -i '1s/^/%global __os_install_post %{nil}/' ~/rpmbuild/SPECS/yggdrasil.spec
+fi
+rpmbuild -bb ~/rpmbuild/SPECS/yggdrasil.spec --target $ARCH
+
+# Build k4e-device-worker rpm
+git clone https://github.com/bardielle/k4e-device-worker.git /home/builder/k4e-device-worker
```

Comment:
at some point we'll have to point back to the main repo. perhaps it should be in this version and referring to personal account could be done locally (or expose account name/branch name as optional parameters)

Reply:
How it can be done locally? we are removing the /home/builder/k4e-device-worker directory

Reply:
We could use env variable for github org and use main repo by default.